helper/testhelpers/consul/consulhelper.go

```diff
@@ -41,21 +41,35 @@ func PrepareTestContainer(t *testing.T, version string) (func(), *Config) {
 		if consulVersion != "" {
 			version = consulVersion
 		} else {
-			version = "1.7.2" // Latest Consul version, update as new releases come out
+			version = "1.11.2" // Latest Consul version, update as new releases come out
 		}
 	}
 	if strings.HasPrefix(version, "1.3") {
 		config = `datacenter = "test" acl_default_policy = "deny" acl_datacenter = "test" acl_master_token = "test"`
 	}
 
-	repo := os.Getenv("CONSUL_DOCKER_REPO")
-	if repo == "" {
-		repo = "consul"
+	name := "consul"
+	isEnterprise := false
+	var envVars []string
+
+	// If running the enterprise container, set the appropriate values below.
+	if strings.HasSuffix(repo, "enterprise") {
+		version += "-ent"
+		name += "-enterprise"
+		isEnterprise = true
+		license, _ := os.LookupEnv("CONSUL_LICENSE")
```

Comment:
It does not work, I tried out the image manually and when there's no license provided it immediately stops running with the output:
```
2022-02-08T18:48:23.081Z [ERROR] agent: Error starting agent: error="license is missing. To add a license, configure "license_path" in your configuration file, use the CONSUL_LICENSE environment variable, or use the CONSUL_LICENSE_PATH environment variable. For a trial license of Consul Enterprise, visit https://consul.io/trial."
2022-02-08T18:48:23.081Z [INFO]  agent: Exit code: code=1
```

However I'm not sure where you'd catch this in the code. I think because the container technically starts correctly it doesn't recognize it as an error. What ends up failing is it timeouts on the endpoint to bootstrap the ACL. It actually tripped me up for a bit when I was passing in the license incorrectly at first. I couldn't find a reference to any issue with the license when running in a debugger.

Reply:
Austin helped clarify my question about this. Instead of trying to see when the docker container stops, I changed it to match what I did in backend_test. If LookupEnv returns with a false as the second value, then it will throw a t.Fatal with the error message "Failed to find enterprise license".